library/core/tests/ptr.rs

```diff
@@ -289,16 +289,17 @@ fn test_const_nonnull_new() {
 }
 
 #[test]
-#[allow(warnings)]
-// Have a symbol for the test below. It doesn’t need to be an actual variadic function, match the
-// ABI, or even point to an actual executable code, because the function itself is never invoked.
-#[no_mangle]
+#[cfg(any(unix, windows))] // printf may not be available on other platforms
+#[allow(deprecated)] // For SipHasher
 pub fn test_variadic_fnptr() {
     use core::hash::{Hash, SipHasher};
     extern "C" {
-        fn test_variadic_fnptr(_: u64, ...) -> f64;
+        // This needs to use the correct function signature even though it isn't called as some
+        // codegen backends make it UB to declare a function with multiple conflicting signatures
+        // (like LLVM) while others straight up return an error (like Cranelift).
+        fn printf(_: *const i8, ...) -> i32;
```

Comment:
```suggestion
        fn printf(_: *const c_char, ...) -> c_int;
```

There are some `cfg(unix)` targets for which `char` is not signed. For example powerpc, riscv64 according to https://github.com/rust-lang/rust/blob/1bfe40d11c3630254504fb73eeccfca28d50df52/library/core/src/ffi/mod.rs.

Reply:
Didn't know the `ffi` module is available in libcore. Fixed.